File: fitness-dimension/service/app/repositories/routine_repository.go
package repositories

import (
	"fitness-dimension/application/routines"
	"fitness-dimension/core/routines/routine"
	valuesObjects "fitness-dimension/core/routines/routine/values-objects"
	"fitness-dimension/service/app/models"

	"github.com/go-pg/pg/v10"
	"github.com/google/uuid"
)

type PgRoutineRepository struct {
	routines.RoutineRepository
	DB *pg.DB
}

func (r PgRoutineRepository) Find(id uuid.UUID) (*routine.Routine, error) {
	var routineModel models.Routine
	err := r.DB.Model().Table("routines").
		Where("id = ?", id.String()).
		Select(&routineModel)

	if err != nil && err != pg.ErrNoRows {
		return nil, err
	}

	if err == pg.ErrNoRows {
		return nil, nil
	}

	id, err = uuid.Parse(routineModel.ID)
	if err != nil {
		return nil, err
	}

	trainingsId, err := r.getTrainings(routineModel)
	if err != nil {
		return nil, err
	}

	return &routine.Routine{
		ID:          valuesObjects.RoutineID{Value: id},
		Name:        valuesObjects.RoutineName{Value: routineModel.Name},
		UserID:      valuesObjects.RoutineUserID{Value: routineModel.UserID},
		Description: valuesObjects.RoutineDescription{Value: routineModel.Description},
		TrainingsID: valuesObjects.RoutineTrainingIDs{Values: trainingsId},
	}, nil

}

func (r PgRoutineRepository) GetAll(userId string) ([]routine.Routine, error) {

	var routineList []models.Routine
	err := r.DB.Model().Table("routines").Where("user_id = ?", userId).Select(&routineList)

	if err != nil && err != pg.ErrNoRows {
		return nil, err
	}

	if err == pg.ErrNoRows {
		return nil, nil
	}

	var routines []routine.Routine
	for _, routineItem := range routineList {

		trainingsId, err := r.getTrainings(routineItem)
		if err != nil {
			return nil, err
		}

		id, err := uuid.Parse(routineItem.ID)
		if err != nil {
			return nil, err
		}

		routines = append(routines, routine.Routine{
			ID:          valuesObjects.RoutineID{Value: id},
			Name:        valuesObjects.RoutineName{Value: routineItem.Name},
			UserID:      valuesObjects.RoutineUserID{Value: routineItem.UserID},
			Description: valuesObjects.RoutineDescription{Value: routineItem.Description},
			TrainingsID: valuesObjects.RoutineTrainingIDs{Values: trainingsId},
		})

	}

	return routines, nil
}

func (r PgRoutineRepository) getTrainings(routineItem models.Routine) ([]uuid.UUID, error) {
	var trainings []models.RoutineTraining
	err := r.DB.Model().Table("routine_training").
		Where("id_routine = ?", routineItem.ID).
		Order("order ASC").
		Select(&trainings)

	if err != nil && err != pg.ErrNoRows {
		return nil, err
	}

	if err == pg.ErrNoRows {
		return nil, nil
	}

	var trainingsId []uuid.UUID
	for _, tId := range trainings {
		i, err := uuid.Parse(tId.TrainingID)
		if err != nil {
			return nil, err
		}

		trainingsId = append(trainingsId, i)
	}

	return trainingsId, nil
}

File: fitness/routine-service/server/src/routine/routine.repository.ts
import { Injectable } from '@nestjs/common';
import { Repository } from 'typeorm';
import { RoutineEntity } from './routine.entity';
import {
  RoutineRepository as IRoutineRepository,
  RoutineDto,
} from 'application';
import {
  Routine,
  RoutineId,
  RoutineName,
  RoutineDescription,
  RoutineUserId,
  RoutineTrainings,
} from 'core';
import { RoutineTrainingEntity } from './routine-training.entity';
import { InjectRepository } from '@nestjs/typeorm';

@Injectable()
export class RoutineRepository implements IRoutineRepository {
  constructor(
    @InjectRepository(RoutineEntity)
    private routineRepository: Repository<RoutineEntity>,
    @InjectRepository(RoutineTrainingEntity)
    private routineTrainingRepository: Repository<RoutineTrainingEntity>,
  ) {}

  async get(uuid: string): Promise<RoutineDto> {
    const routine = await this.routineRepository.findOne({
        where: { id: uuid },
        relations: {
          trainings: true,
        },
      }),
      dto: RoutineDto = {
        id: routine.id,
        name: routine.name,
        description: routine.description,
        userId: routine.userId,
        trainings:
          routine.trainings?.map((training) => training.idTraining) || [],
      };

    return dto;
  }

  async getAll(userId: string): Promise<RoutineDto[]> {
    const routines = await this.routineRepository.find({
        where: { userId: userId },
        relations: {
          trainings: true,
        },
      }),
      dtos: RoutineDto[] = routines.map((routine) => {
        const dto: RoutineDto = {
          id: routine.id,
          name: routine.name,
          description: routine.description,
          userId: routine.userId,
          trainings: routine.trainings.map((training) => {
            return training.idTraining;
          }),
        };
        return dto;
      });
    return dtos;
  }

  async getRoutine(uuid: string): Promise<Routine> {
    const routineEntity = await this.routineRepository.findOneBy({
        id: uuid,
      }),
      trainings = await this.routineTrainingRepository.find({
        where: { idRoutine: uuid },
        order: { order: 'ASC' },
      }),
      routine = new Routine(
        new RoutineName(routineEntity.name),
        new RoutineUserId(routineEntity.userId),
        new RoutineDescription(routineEntity.description),
        new RoutineTrainings(trainings.map((training) => training.idTraining)),
        new RoutineId(routineEntity.id),
      );

    console.log(routine.id, uuid);

    return routine;
  }
}
